addons/payment/utils.py

```diff
@@ -22,9 +24,15 @@ def generate_access_token(*values):
     :return: The generated access token
     :rtype: str
     """
-    token_str = '|'.join(str(val) for val in values)
-    access_token = hmac_tool(request.env(su=True), 'generate_access_token', token_str)
-    return access_token
+    try:
+        environment = env(su=True) if env else request.env(su=True)
+    except RuntimeError:
+        _logger.exception('No suitable environment available to generate payment access token')
```

Comment:
I don't think this is super useful; it's going to crash and dump a traceback mentioning the call to `generate_access_token` anyway, so it doesn't add information for debugging.